ethereumj-core/src/main/java/org/ethereum/sync/SyncManager.java

```diff
@@ -204,7 +235,7 @@ private void blockRetrieveLoop() {
                         // to get more chances to receive block body promptly
                         for (BlockHeaderWrapper blockHeaderWrapper : bReq.getBlockHeaders()) {
                             Channel channel = pool.getByNodeId(blockHeaderWrapper.getNodeId());
-                            if (channel != null) {
+                            if (channel != null && channel.isIdle()) {
```

Comment:
Sometimes cause peer drop with reputation reduce.
Actually we have no guarantee in multi-thread env on the next few lines but had not ever run it in.